src/Raven.Server/Config/Categories/IndexingConfiguration.cs

```diff
@@ -95,6 +95,8 @@ public IndexingConfiguration(RavenConfiguration root)
             
             EncryptedTransactionSizeLimit = defaultEncryptedTransactionSizeLimit;
             MaxAllocationsAtDictionaryTraining = defaultMaxAllocationsAtDictionaryTraining;
+
+            MaxMemoizationSize = Size.Max(new Size(512, SizeUnit.Megabytes), totalMem / 10);
```

Comment:
`totalMem / 10` - can it be rounded to power of 2?

Reply:
Also I'd recommend to have some upper limit. On systems with high amount of installed memory (e.g. 256GB) that could be really high.

Reply:
The problem with powers of two is that the gaps between numbers after 2GB are large. I think we can use `Bits.NextAllocationSize`, which aligns the memory to 64MB.

What do you suggest should be the upper limit?

Reply:
Don't we already have a limit of 2GB here?

Reply:
Yes, that is an issue when sorting over 250M elements, right?

Reply:
Does that even _make sense_ ?